### PR TITLE
GuidedTours: Implement mobile styling & positioning

### PR DESCRIPTION
**Pull request by mcsf**
Implements #4642

### TODO
- [x] Adjust width on mobile
- [x] Deal with Themes step, which doesn't fit vertically
- [x] Small visual tweaks (shadow, margin?)
- [x] Test on actual mobile devices
- [x] Make meaningful commit message(s)

### CODE REVIEW

**Comment by mcsf** [comment]
@ehg @lsinger: should be ready for review.

**Comment by lsinger** [comment]
Awesome, this is great to see! 

I wonder a bit about the scrolling. The sudden jump isn't really noticable, and so feels kind of disorienting when you're in a different place suddenly.

**Comment by lsinger** [comment]
The sidebar is scrolled a bit even when Themes and the step below it would be visible.

**Comment by lsinger** [comment]
The final step doesn't seem to be centered anymore in Desktop-sized windows.

**Comment by lsinger** [comment]
If the browser window is tall but has a mobile-sized width (~33% the width and 100% the height of a MBP 13"), the Themes menu item is tried to be scrolled into view, but not really because there isn't enough to scroll because the window is so tall. It is visible, but the step dialog is placed over it. `getStepPosition` or `dialogPositioners` probably need some tweaking to take that into account.

**Comment by mcsf** [comment]
> I wonder a bit about the scrolling. The sudden jump isn't really noticable, and so feels kind of disorienting when you're in a different place suddenly.

Yep. Thanks for having a go and for working on animating the scroll!

One small concern I have regards `DIALOG_HEIGHT`; can we safely assume a magic value there? I mean, we can, but it will be an estimate for each device.

Complementary idea: if, after rendering, we detect that the buttons in the step dialog are not contained in the viewport, we wait a couple of seconds and move to the next step. ❓

**Comment by lsinger** [comment]
> One small concern I have regards DIALOG_HEIGHT; can we safely assume a magic value there? I mean, we can, but it will be an estimate for each device.

Yes, it's a wip assumption. But maybe it's good enough for most cases, and for the edge cases we move it up "manually" after a few seconds after a check has found that it's partly off-screen? 

> if, after rendering, we detect that the buttons in the step dialog are not contained in the viewport, we wait a couple of seconds and move to the next step

That sounds like a reasonable fail-safe. OTOH we'd have to click buttons that would alter the state of the UI for the user without their doing. Suddenly the sidebar pops up. That sounds confusing also. Maybe just cancel the tour if a step is found to be rendered off-screen--wdyt? 

We shouldn't cancel the tour if they just scrolled away to look at other things, though. Maybe only if it's in a position that cannot be scrolled to, such as (-100, -100)? Then again, we already guard against coordinates like these.

File: client/guidestours/positioning.js [hunk 1]
```diff
@@ -75,13 +75,31 @@ const bullseyePositioners = {
 	middle: () => ( {} ),
 };
 
-export function getStepPosition( { placement = 'center', target } ) {
+export function getStepPosition( { placement = 'center', targetSlug } ) {
+	const target = query( '[data-tip-target="' + targetSlug + '"]' )[0];
```
Comment (ehg) [inline client/guidestours/positioning.js]: Does this mean we can get rid of tipTargets in `guidestours/index`? :o

Comment (lsinger) [inline client/guidestours/positioning.js]: Good point! Seems likely.

Comment (mcsf) [inline client/guidestours/positioning.js]: Seems a lot sturdier to grab `target` in here, definitely!

Comment (lsinger) [inline client/guidestours/positioning.js]: In `GuidesTours:quit` we still use `currentTarget` to remove the overlay on quit (step-component-agnostic). Not sure we want to figure out how to get rid of that right now.

**Comment by lsinger** [comment]
Rebased.

**Comment by ehg** [comment]
🚢  !

**Pull request merged**